Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -938,15 +937,18 @@ ThrowCompletionOr<NonnullGCPtr<Module>> VM::resolve_imported_module(ScriptOrModu
 
     dbgln_if(JS_MODULE_DEBUG, "[JS MODULE] reading and parsing module {}", filename);
 
-    auto file_or_error = Core::DeprecatedFile::open(filename, Core::OpenMode::ReadOnly);
+    auto file_or_error = Core::File::open(filename, Core::File::OpenMode::Read);
 
     if (file_or_error.is_error()) {
         return throw_completion<SyntaxError>(ErrorType::ModuleNotFound, module_request.module_specifier);
     }
 
     // FIXME: Don't read the file in one go.
-    auto file_content = file_or_error.value()->read_all();
-    StringView content_view { file_content.data(), file_content.size() };
+    auto file_content_or_error = file_or_error.value()->read_until_eof();
+    if (file_content_or_error.is_error()) {
+        return throw_completion<SyntaxError>(ErrorType::OutOfMemory, module_request.module_specifier);
```

Comment:
Shouldn't it be an `InternalError` and not a syntax one?

Reply:
I have no idea! The argument in my head was that "insanely large file == syntax error". Yup, I'll change it to InternalError then.